### ISSUE TITLE
PanelMenu - Repeating menu items

### ISSUE THREAD

**Issue by jalley3** [primefaces/primeng#15796]
### Describe the bug

https://github.com/primefaces/primeng/issues/14373 ... basically this bug is back... see my comment at the end for current pictures ...note this is working in 17.16

### Environment

version 17.17 and 17.18

### Angular version

17.3.4

### PrimeNG version

17.17 

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.10

### Steps to reproduce the behavior

![image](https://github.com/primefaces/primeng/assets/17911817/b35ecf0f-9100-4d77-a899-b295557abd01)

### Expected behavior

![image](https://github.com/primefaces/primeng/assets/17911817/047118ee-7945-42db-89d3-55195c8edc5f)

**Comment by jalley3** [comment]
@mehmetcetin01140 i believe it is related to your changes

**Comment by guaido79** [comment]
I found thats uncommon behavior too. 
I see it's due to the routerLink in the MenuItem. If you remove it it come back to single menu per item. 
To fix my code i add another property in the any property of MenuItem and bind it in the menu template.

**Comment by jalley3** [comment]
remove the routerlink? those are supposed to be links to other pages so i need the router links @guaido79

**Comment by guaido79** [comment]
> remove the routerlink? those are supposed to be links to other pages so i need the router links @guaido79

Yes, sure, I give just some more information to find the problem. 

in the example of the showcase on Stackblitz **_routerLink_** it's not used:

```typescript
items: MenuItem[];

    constructor(private router: Router) {}

    ngOnInit() {
        this.items = [
            {
                label: 'Router',
                icon: 'pi pi-palette',
                items: [
                    {
                        label: 'Installation',
                        icon: 'pi pi-eraser',
                        route: '/installation'
                    },
                    {
                        label: 'Configuration',
                        icon: 'pi pi-heart',
                        route: '/configuration'
                    }
                ]
            },
            {
                label: 'Programmatic',
                icon: 'pi pi-link',
                command: () => {
                    this.router.navigate(['/installation']);
                }
            },

``` 


```html
<p-panelMenu [model]="items" styleClass="w-full md:w-20rem">
        <ng-template pTemplate="item" let-item>
            <ng-container *ngIf="item.route; else urlRef">
                <a [routerLink]="item.route" class="flex align-items-center 
                    cursor-pointer text-color px-3 py-2">
                    <span [class]="item.icon"></span>
                    <span class="ml-2 text-color">{{ item.label }}</span>
                </a>
            </ng-container>
```

**Comment by jalley3** [comment]
@guaido79 thank you however MenuItem does not even contain a route property (https://primeng.org/dock#api.dock.interfaces.MenuItem ) which breaks a lot of our strongly typed methods and means redoing our entire site's navigation that is currently set up to use RouterLink in the MenuItem. Its frustrating because it just worked in v17.16

**Comment by cetincakiroglu** [comment]
Hi @jalley3 ,

Could you please share a stackblitz reproducer?

**Comment by SippieCup** [comment]
I don't believe this has anything to do with routerLink or the properties in question besides the fact that the documentation is wrong.

I think it is because if a custom template is specified, it will show both the custom template and then the default template.

https://github.com/primefaces/primeng/blob/d10b289c0e5904248e58fd043d23dc79870777f0/src/app/components/panelmenu/panelmenu.ts#L872


The routerLink thing is a red herring because it will only duplicate content when the *ngIf matches.

**Comment by SippieCup** [comment]
Reproduction on v18:
https://stackblitz.com/~/github.com/SippieCup/panelMenu-duplication

**Comment by SippieCup** [comment]
@cetincakiroglu I believe it is from this `<a>` tag being outside of the container that is checking for itemTemplate.

https://github.com/primefaces/primeng/blob/c89e42c654253022d3c21b6e0c8580cec676ea63/src/app/components/panelmenu/panelmenu.ts#L872

This causes it to render both the item template, and the additional link.

**Comment by lucasfnds** [comment]
In version v19.0.2 the problem still occurs, does anyone have a solution or a deadline for when the correction will be released?

**Comment by Cryxto** [comment]
in v20 it is stil occur
